### PR TITLE
tests: mutex: convert legacy test to ztest

### PR DESCRIPTION
**Pull request by punitvara**
Make legacy test case use of ztest apis to support
ztest framework.

Signed-off-by: Punit Vara <punit.vara@intel.com>

### CODE REVIEW

**Comment by punitvara** [comment]
This is just for testing purpose to get reviewed my code. Please do not review this PR until I send fully updated code.

File: tests/kernel/mutex/mutex/src/mutex.c [hunk 1]
```diff
@@ -241,7 +242,7 @@ extern void thread_12(void);
 
 void user_main(void)
 {
-	int rv;
```
Comment (andrewboie) [inline tests/kernel/mutex/mutex/src/mutex.c]: why was this indented

File: tests/kernel/mutex/mutex/src/mutex.c [hunk 2]
```diff
@@ -428,3 +422,14 @@ K_THREAD_ACCESS_GRANT(THREAD_07, &mutex_3);
...
 K_THREAD_ACCESS_GRANT(THREAD_11, &mutex_3);
+
+/*test case main entry*/
+void test_main(void)
+{
+	k_thread_access_grant(k_current_get(), &private_mutex,
+			      &mutex_1, &mutex_2, &mutex_3, &mutex_4,
+			      &thread_12_thread_data, &thread_12_stack_area,
+			      NULL);
+	ztest_test_suite(test_mutex, ztest_unit_test(testing_mutex));
```
Comment (andrewboie) [inline tests/kernel/mutex/mutex/src/mutex.c]: can you use ztest_user_unit_test() instead of dropping to user mode later?

**Comment by punitvara** [comment]
Depends on #4903 This will pass shipable after that get merged.

File: tests/kernel/mutex/mutex/src/mutex.c (hunk 2 above)
Comment (punitvara) [reply tests/kernel/mutex/mutex/src/mutex.c]: Thanks that helped.

**Review by nashif — APPROVED** [review]

**Pull request merged**